lib/frederick_api/v2/helpers/paginator.rb

```diff
@@ -17,15 +17,25 @@ def all_records
 
           (total_pages - current_page).times do
             first_resource.class.with_headers(first_resource.custom_headers) do
-              current_result_set = current_result_set ? current_result_set.pages.next : self.result_set.pages.next
+              current_result_set = next_result_set(current_result_set)
               raise 'next link not found' unless current_result_set
               results.push(*current_result_set.to_a)
             end
           end
-
           results
         end
 
+        # refetching the next link if no response is found due to elastic cache unknown beheviour.
+        # bug link: https://dev.azure.com/mindbody/Marketing%20Products/_workitems/edit/1101861
+        def next_result_set(current_result_set)
```

Comment:
I think we should make this a private method and add to the tests for all_records - add tests for when the current result set is empty.